commands/delete-repository.ts

```diff
@@ -0,0 +1,167 @@
+import * as vscode from "vscode";
+import request = require('request-promise');
+import * as azureUtils from '../explorer/utils/azureUtils';
+import { SubscriptionModels } from 'azure-arm-resource';
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import { AzureCredentialsManager } from '../utils/AzureCredentialsManager';
+import { AzureRepositoryNode } from '../explorer/models/AzureRegistryNodes';
+const teleCmdId: string = 'vscode-docker.deleteRepository';
+
+
+/**
+ * function to delete an Azure repository and its associated images
+ * @param context : if called through right click on AzureRepositoryNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ */
+export async function deleteRepository(context?: AzureRepositoryNode) {
+
+    let azureAccount = await AzureCredentialsManager.getInstance().getAccount();
+    if (!azureAccount) {
+        return;
+    }
+
+    if (azureAccount.status === 'LoggedOut') {
+        return;
+    }
+    let registry: Registry;
+    let subscription: SubscriptionModels.Subscription;
+    let repoName: string;
+    let username: string;
+    let password: string;
+    if (!context) {
+        //first get desired registry
+        let registries = await AzureCredentialsManager.getInstance().getRegistries();
+        let reg: string[] = [];
+        for (let i = 0; i < registries.length; i++) {
+            reg.push(registries[i].name);
+        }
+        let desired = await vscode.window.showQuickPick(reg, { 'canPickMany': false, 'placeHolder': 'Choose the Registry from which your desired repository exists' });
```

Comment:
>'Choose the Registry from which your desired repository exists [](start = 100, length = 62)

What is the guidelines for putting all these messages together. Asking because we want to consider localization if needed.